web/src/service-worker/index.ts

```diff
@@ -0,0 +1,161 @@
+/// <reference types="@sveltejs/kit" />
+/// <reference no-default-lib="true"/>
+/// <reference lib="esnext" />
+/// <reference lib="webworker" />
+
+const sw = globalThis as unknown as ServiceWorkerGlobalScope;
+import { build, files, version } from '$service-worker';
+
+const pendingLoads = new Map<string, AbortController>();
+const useCache = true;
+// Create a unique cache name for this deployment
+const CACHE = `cache-${version}`;
+
+const APP_RESOURCES = [
+  ...build, // the app itself
+  ...files, // everything in `static`
+];
+
+sw.addEventListener('install', (event) => {
+  event.waitUntil(sw.skipWaiting());
+  // Create a new cache and add all files to it
+  event.waitUntil(addFilesToCache());
+});
+
+sw.addEventListener('activate', (event) => {
+  event.waitUntil(sw.clients.claim());
+  // Remove previous cached data from disk
+  event.waitUntil(deleteOldCaches());
+});
+
+sw.addEventListener('fetch', (event) => {
+  // ignore POST requests etc
+  if (event.request.method !== 'GET') return;
```

Comment:
```suggestion
  if (event.request.method !== 'GET') {
    return;
  }
```

Reply:
That's caught by a linting rule as well. Generally all failing actions should be addressed.